swabbie-redis/src/main/kotlin/com/netflix/spinnaker/swabbie/redis/RedisResourceTrackingRepository.kt

```diff
@@ -34,76 +35,107 @@ class RedisResourceTrackingRepository(
   private val clock: Clock
 ) : ResourceTrackingRepository {
 
+  private val SINGLE_RESOURCES_KEY = "{swabbie:resource}"
+  private val SOFT_DELETE_KEY = "{swabbie:softdeletes}"
+  private val DELETE_KEY = "{swabbie:deletes}"
+
+  private val log = LoggerFactory.getLogger(javaClass)
   private val redisClientDelegate: RedisClientDelegate = redisClientSelector.primary("default")
 
   override fun find(resourceId: String, namespace: String): MarkedResource? {
-    "$namespace:$resourceId".let { key ->
-      return redisClientDelegate.withCommandsClient<String> { client ->
+    val key = "$namespace:$resourceId"
+    return try {
+      redisClientDelegate.withCommandsClient<String> { client ->
         client.hget(SINGLE_RESOURCES_KEY, key)
-      }.let { json ->
-          objectMapper.readValue(json)
-        }
+      }.let { objectMapper.readValue(it) }
+    } catch (e: IllegalStateException) {
+      // resource does not exist
+      null
```

Comment:
Mmm... this doesn't sound right. What is throwing ISE? The redis client or the object mapper? If it's the object mapper, the resource exists, but it's malformed.

Reply:
Oh I think I just need the handy `?` because the object mapper was complaining the value was null. Not fully used to the awesomeness of Kotlin.